common/Perf/Azure.Test.Perf/PerfProgram.cs

```diff
@@ -174,6 +178,48 @@ private static async Task Run(Type testType, PerfOptions options)
             PrintAssemblyVersions(testType);
         }
 
+        private static void ConfigureThreadPool(PerfOptions options)
+        {
+            if (options.MinWorkerThreads.HasValue || options.MinCompletionPortThreads.HasValue)
+            {
+                ThreadPool.GetMinThreads(out var minWorkerThreads, out var minCompletionPortThreads);
+                ThreadPool.SetMinThreads(options.MinWorkerThreads ?? minWorkerThreads,
+                    options.MinCompletionPortThreads ?? minCompletionPortThreads);
+            }
+
+            if (options.MaxWorkerThreads.HasValue || options.MaxCompletionPortThreads.HasValue)
+            {
+                ThreadPool.GetMaxThreads(out var maxWorkerThreads, out var maxCompletionPortThreads);
+                ThreadPool.SetMaxThreads(options.MaxWorkerThreads ?? maxWorkerThreads,
+                    options.MaxCompletionPortThreads ?? maxCompletionPortThreads);
+            }
+        }
+
+        private static void PrintEnvironment()
+        {
+            Console.WriteLine("=== Environment ===");
+
+            Console.Write("Configuration: ");
+#if DEBUG
+            Console.WriteLine("Debug");
+#elif RELEASE
+            Console.WriteLine("Release");
+#endif
+
+            Console.WriteLine($"GCSettings.IsServerGC: {GCSettings.IsServerGC}");
+
+            Console.WriteLine($"Environment.ProcessorCount: {Environment.ProcessorCount}");
```

Comment:
worth adding `System.Environment.Is64BitProcess`, got surprised by this lately...